doc/manage-duniverse.mld

```diff
@@ -0,0 +1,213 @@
+the regular opam workflow? Compared to regular opam usage using opam-monorepo
+brings lockfiles and a duniverse folder into the repository. There are multiple
+ways to handle this, with multiple advantages and disadvantages. This document
+explains the considerations behind dealing with each of those, along with our
+recommendations.
+
+The document mentions Git for source control as it currently is the most
+popular choice, but similar considerations are valid for most other version
+control systems as well.
+
+{1 Methods}
+
+We assume the opam file is committed to the repository. Running [opam monorepo
+lock] will create a [.locked] opam file in addition to the original opam file.
+Calling [opam monorepo pull] will create a [duniverse] folder into which it
+will unpack the sources of the packages that are mentioned in the lockfile.
+
+{2 Not locked}
+
+While this is not a [opam-monorepo] workflow per se, projects that do not use
+[opam-monorepo] can still be used with [opam-monorepo] and can bring some
+advantages to the project.
+
+In this workflow such every contributor is responsible for creating and
+maintianing their own lockfile and duniverse.
+
+Advantages include:
+
+{ul
+{- No opt-in to [opam-monorepo] required, every developer can chose to use it
+   or not}
+{- Easier management of dependencies over dependencies installed in opam
+   switches}
+{- No merge conflicts when multiple collaborators change versions}
+{- Having the dependencies built by Dune makes cross-compilation easier}
+}
+
+If comes however with some caveats:
+
+{ul
+{- Everyone might end up with different packages in their duniverse, thus low
+   reproducability}
+{- Not self-contained}
+}
+
+{3 Adoption}
+
+Using this workflow is very simple
+
+{[
+$ opam monorepo lock
+$ opam monorepo pull
+]}
+
+This will lock the versions of packages into a lock file and will download the
+sources into the [duniverse] folder. Thus a subsequent [dune build] will also
+build the dependencies.
+
+To make sure the files don't get committed by accident, it is useful to add the
+following excludes to the [.gitignore] file:
+
+{[
+*.opam.locked
+duniverse/
+]}
+
+{2:lockfile-in-git Lockfile in Git}
+
+Another approach is very similar to the regular opam workflow, but in addition
+to having the [.opam] file committed to the repository, also the lockfile that
+[opam monorepo lock] generates is added.
+
+Advantages include:
+
+{ul
+{- The project repository only grows by one file, thus easy to adopt}
+{- The packages to be installed are reproducible between different machines
+   and contributors}
+{- Easier management of dependencies over dependencies installed in opam
+   switches}
+{- Having the dependencies built by Dune makes cross-compilation easier}
+}
+
+Disadvantages might be
+
+{ul
+{- Everyone who wants to build the dependencies with Dune needs to have
+   [opam-monorepo] installed to pull the dependencies first}
+{- Not self-contained}
```

Comment:
Maybe say something about the CI which should now support testing both the locked and non-locked versions?

Reply:
As far as I know the CI has a single opam-monorepo workflow which is: if there is a monorepo lockfile, `pull` and then build.

It shouldn't be too hard to eventually detect that the duniverse is present and skip the first bit but it's not doing it at the moment.

Reply:
To clarify: we should mention the CI indeed but also make sure users know exactly what it supports.

Reply:
Also, the two items in this list derive from the same thing: dependencies are not packaged in the repository. I think they should be merged into one!